src/renderers/webxr/WebXRManager.js

```diff
@@ -232,18 +235,16 @@ class WebXRManager extends EventDispatcher {
 				session.addEventListener( 'end', onSessionEnd );
 				session.addEventListener( 'inputsourceschange', onInputSourcesChange );
 
-				const attributes = gl.getContextAttributes();
-
 				if ( attributes.xrCompatible !== true ) {
 
 					await gl.makeXRCompatible();
 
 				}
 
-				if ( session.renderState.layers === undefined ) {
+				if ( ( session.renderState.layers === undefined ) || ( gl instanceof WebGLRenderingContext ) ) {
```

Comment:
Hmm, when would `gl` not be an instance of `WebGLRenderingContext`?

Reply:
When it's a WebGL2 rendering context? Or is this a wrong way of checking?